src/checks/stylelint.ts

```diff
@@ -0,0 +1,40 @@
+import { execaCommand } from 'execa';
+
+export const title = 'Stylelint';
+
+export default async function stylelintCheck() {
+  try {
+    await execaCommand(
+      'pnpm stylelint **/*.{css,scss,less,js,tsx} --max-warnings 0',
+    );
+  } catch (error) {
+    const { stdout } = error as { stdout: string };
+    const lines = stdout.split('\n');
+
+    // If no Stylelint problems detected, throw the error
+    if (
+      !/^\d+ problems? \(\d+ errors?, \d+ warnings\)$/.test(
+        lines[lines.length - 2]!,
+      )
+    ) {
+      throw error;
+    }
+
+    throw new Error(
+      `Stylelint problems found in the following files:
+
+      ${lines
+        .filter((line) => {
+          return (
+            line !== '' &&
+            !line.match(/^\d+ problems? \(\d+ errors?, \d+ warnings\)|\d+:\d+/)
+          );
```

Comment:
I guess this can probably be simplified, similar to the ESLint check `.filter()`

Reply:
https://github.com/upleveled/preflight/pull/377/commits/f0dbdb404c6b73fdfdc25cc8870d21dad9c7fef7